mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/ChessFacade.kt

```diff
@@ -199,7 +208,28 @@ private data class StoreMatch(
       }
 
   override suspend fun update(game: Game) {
-    store.collection("games").document(id).update { this["moves"] = game.toAlgebraicNotation() }
+    val chess = store.collection("games").document(id).get<ChessDocument>()
+
+    store.collection("games").document(id).update {
+      this[FieldPath(listOf("metadata", "status"))] =
+          when (game.nextStep) {
+            NextStep.Stalemate -> "stalemate"
+            is NextStep.MovePiece -> null
+            else ->
+                if ((game.nextStep as NextStep.Checkmate).winner == Color.Black) "blackWon"
+                else "whiteWon"
+          }
+
+      if (chess?.blackId == user?.uid && chess?.metadata?.blackName != user?.name) {
+        this[FieldPath(listOf("metadata", "blackName"))] = user?.name
+      }
+
+      if (chess?.whiteId == user?.uid && chess?.metadata?.whiteName != user?.name) {
+        this[FieldPath(listOf("metadata", "whiteName"))] = user?.name
+      }
+
```

Comment:
I think you should be updating the last updated timestamp for the chess document here.

Reply:
yes